src/main/java/com/yelp/nrtsearch/server/luceneserver/doc/LoadedDocValues.java

```diff
@@ -333,6 +341,44 @@ public SortedDateTimes(SortedNumericDocValues docValues) {
     }
   }
 
+  public static final class ObjectJsonDocValues extends LoadedDocValues<Map<String, Object>> {
+    private final BinaryDocValues docValues;
+    private List<Map<String, Object>> value;
+
+    public ObjectJsonDocValues(BinaryDocValues docValues) {
+      this.docValues = docValues;
+    }
+
+    @Override
+    public void setDocId(int docID) throws IOException {
+      if (docValues.advanceExact(docID)) {
+        String jsonString = STRING_DECODER.apply(docValues.binaryValue());
+        value = gson.fromJson(jsonString, List.class);
+      } else {
+        value = null;
+      }
+    }
+
+    @Override
+    public Map<String, Object> get(int index) {
+      if (value == null) {
+        throw new IllegalStateException("No doc values for document");
+      }
+      return value.get(index);
```

Comment:
It would be good to also check the index range and throw IndexOutOfBoundsException